lib/src/iterable_extensions.dart

```diff
@@ -592,6 +592,8 @@ extension IterableNullableExtension<T extends Object> on Iterable<T?> {
 /// since doubles require special handling of [double.nan].
 extension IterableNumberExtension on Iterable<num> {
   /// A minimal element of the iterable, or `null` it the iterable is empty.
+  ///
+  /// If an element is NaN, it is considered to be a minimal element.
```

Comment:
Hmm, pretty over-engineered code, now that I look at it again.
```dart
  num? get minOrNull {
    var empty = true;
    var min = double.infinity;
    for (var value in this) {
       empty = false;
       if (value < min) min = value;
    }
    if (empty) return null;
    return min;
  }
```
seems much more optimizable.

(Maybe I should make a benchmark :grin:)